### ISSUE TITLE
🐛 BUG: Two CSS bugs with `astro dev` and TailwindCSS

### ISSUE THREAD

**Issue by SKFrozenCloud** [withastro/astro#3494]
### What version of `astro` are you using?

1.0.0-beta.38

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

npm and yarn

### What operating system are you using?

Linux

### Describe the Bug

There are two bugs both related to when you import base tailwind yourself with
`tailwind({
      config: {
        applyBaseStyles: false,
      },
    })`

1. When setting to manually import Tailwind into each page/layout and manually importing the `tailwind.css` file will cause the importation of ALL CSS styles from all other pages when using `astro dev`. This does not happen if you use `astro build && astro preview`.

2. When setting to manually import Tailwind into each page/layout  and instead of importing a `tailwind.css` file you put `<style is:global>
		@tailwind base;
		@tailwind components;
		@tailwind utilities;
	</style>` directly into the layout/page file, it will not import all Tailwind directives such as `md:` (example: `md:w-1/4` will not work but `w-1/4` will work)

Workaround is to import the `tailwind.css` file and just ignore the conflicting/other css styles until you test with `astro build && astro preview`.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-8x99bh-bud4qs

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by tony-sull** [comment]
Confirmed this reproduces for me locally as well, and only when Tailwind is being used.  Thanks for the clear minimal reproduction, @SKFrozenCloud!

This might end up being a very ...interesting... one to fix 😆 Somehow the style used by a markdown page is being added as a module dependency to a different page, very curious how that's even related to whether Tailwind is being used 🤷

**Comment by SKFrozenCloud** [comment]
Interesting. I don't know much about Vite and building tools in general otherwise I would be happy to find what caused the bug.

The weird part is that it works when you do `astro build && astro preview`.
Perhaps the issue is related to SSR or HMR together with Tailwind?

**Comment by tony-sull** [comment]
Yep it's likely coming from either HMR or from how Astro's dev server processes styles internally

I'm actually wrapping up a big HMR improvement today so I'll test this bug again once the HMR update is released to see if that happened to fix this as well

**Comment by thomfoolery** [comment]
Can confirm that I am experiencing this issue too

**Comment by tony-sull** [comment]
Well this one is every bit as crazy as I expected!

#2 is an easier answer, tailwind should be imported from a `.css` file rather than in an inline `<style>` block.  Vite recommends that solution in their [docs](https://tailwindcss.com/docs/guides/vite) and we're following the leader here

#1 is a bit crazier.  In the original repro there's a custom `tailwind.config.cjs` file that includes `.md` files for Tailwind compilation

```
module.exports = {
    content: ['./src/**/*.{md,...}
}
```

This is actually what's triggering the bug - when `pages/index.astro` is being built Tailwind is scanning the markdown file, which actually means Astro renders the markdown page including the `DocsLayout` component.  Everything in the markdown page is then considered a dependency of `index.astro`

Basically, to fix this just remove the `md` extends from Tailwind's `content` option

**Comment by tony-sull** [comment]
Well this one ended up being every bit as crazy as I expected!

#2 is easy enough to answer - `@tailwind` should always be done in a `.css` file and imported in frontmatter.  This matches [Tailwinds's docs](https://tailwindcss.com/docs/guides/vite) for using Tailwind in Vite

#1 is the crazy part.  I've so far tracked it down to the reproduction's custom `tailwind.config.cjs` file.  It includes `md` as a file extension for Tailwind's `content` option

When `index.astro` is being built in dev, it's actually triggering the unrelated markdown pages to be built and they're then treated as dependencies to `index.astro`.  Removing the `md` extension from the `content` config fixes this, but that would mean any Tailwind classes in the markdown page would be ignored

The simple solution that may unblock you is to simply remove the `.md` extension from your Tailwind config.  Unless your project is using HTML with and Tailwind classes that wouldn't be a problem, but that limitation is something we'll need to dig further into

**Comment by SKFrozenCloud** [comment]
But there is no `main.js` to import the `tailwind.css` file. I just imported it in the astro layout file.

**Comment by SKFrozenCloud** [comment]
Updated the astro package from version `1.0.0-beta.44` to `1.0.0-beta.69` and it broke.

https://github.com/withastro/astro/issues/3910

**Comment by SKFrozenCloud** [comment]
Issue is no longer there
